UNLV-CS472-672/2024-S-GROUP5-Munch
# Add Like Button Functionality

## Description / Changes Made
- Added functionality to the like button for posts that have a like button (non Yelp)
- Like button sends API request to like and unlike
- Like button now has a like counter next to it
- Added an example like button in the Feed tab
  - This like button is hardcoded to a specific post
  - This exists to test the like button without needing the logged in user to be following any other user
  - This can be removed in future PRs with no worry
 - Visual changes to the like button will come in a future PR
 - Changes made to friends.tsx were authored by @noodaj in order to make the Post ID accessible

## Portions of this code that utilized AI generation
No AI was used to generate this code since existing documentation and collaboration with other team members was sufficient. 

## How to Test

1. Start the Firebase Emulator and backend server
2. Start the frontend server and run your device emulator
3. Log into a user
4. Disable the middleware 
5. Use Postman to follow the "munch" user
6. Reenable the middleware
7. Relog into the user you were using
8. Navigate to the Feed tab
9. Press the like button in the upper left corner
10. You should see the API call in the backend terminal and the like count update on the device emulator.
11. Navigate to the Friends tab
12. You should see a list of posts. Interact with the like button on a post
13. You should see the number of likes change. If you check the Firebase Emulator site and you search for the post, you should also see the number of likes change

## Checklist
